hpoflow/optuna_transformers.py

```diff
@@ -90,23 +91,42 @@ def setup(self, args, state, model):
                 )
         self._initialized = True
 
-    def on_train_begin(self, args, state, control, model=None, **kwargs):
+    def on_train_begin(
+        self,
+        args: TrainingArguments,
+        state: TrainerState,
+        control: TrainerControl,
+        model=None,
+        **kwargs,
+    ) -> None:
         """
+        Event called at the beginning of training.
+
         Call setup if not yet initialized.
         """
         if not self._initialized:
             self.setup(args, state, model)
 
-    def on_log(self, args, state, control, logs, model=None, **kwargs):
+    def on_log(
+        self,
+        args: TrainingArguments,
+        state: TrainerState,
+        control: TrainerControl,
+        logs: Dict[str, Number],
+        model=None,
+        **kwargs,
+    ):
         """
+        Event called after logging the last logs.
+
         Log all metrics from Transformers logs as MLflow metrics at the appropriate step.
         """
         if not self._initialized:
             self.setup(args, state, model)
         if state.is_world_process_zero:
             metrics_to_log = dict()
             for k, v in logs.items():
-                if isinstance(v, (int, float)):
+                if isinstance(v, (int, float)):  # TODO: remove or change to Number?
```

Comment:
@twolffpiggott what do you think? Can we remove this check since we use typing? Or change the
check to check for Number?